silkrpc/commands/debug_api.cpp

```diff
@@ -390,9 +390,10 @@ boost::asio::awaitable<void> DebugRpcApi::handle_debug_trace_block_by_number(con
         const auto block_with_hash = co_await core::read_block_by_number(*context_.block_cache(), tx_database, block_number);
 
         debug::DebugExecutor executor{*context_.io_context(), tx_database, workers_, config};
-        const auto debug_traces = co_await executor.execute(block_with_hash.block);
+        const auto debug_trace_result = co_await executor.execute_block(block_with_hash.block);
```

Comment:
I suggest a different approach here:
- do not add the new method `DebugExecutor::execute_block`, but use the existing `DebugExecutor::execute`
- create a `DebugTraceResultList` object containing `debug_traces` returned by `execute`
- add `to_json` serialisation method for `DebugTraceResultList` and handle the additional `result` field there

Reply:
Ok